unleash-yggdrasil/src/lib.rs

```diff
@@ -89,16 +93,49 @@ fn build_segment_map(segments: &Option<Vec<Segment>>) -> HashMap<i32, Segment> {
         .unwrap_or_default()
 }
 
+fn compile_variant_rule(
+    toggle: &ClientFeature,
+    segment_map: &HashMap<i32, Segment>,
+) -> Option<Vec<(RuleFragment, Vec<CompiledVariant>)>> {
+    let variant_rules: Option<Vec<(RuleFragment, Vec<CompiledVariant>)>> =
+        build_variant_rules(&toggle.strategies.clone().unwrap_or_default(), segment_map)
+            .iter()
+            .map(|(rule_string, strategy_variants, stickiness)| {
+                let compiled_rule: Option<RuleFragment> = compile_rule(rule_string).ok();
+                compiled_rule.map(|rule| {
+                    (
+                        rule,
+                        strategy_variants
+                            .iter()
+                            .map(|strategy_variant| CompiledVariant {
+                                name: strategy_variant.name.clone(),
+                                weight: strategy_variant.weight,
+                                stickiness: Some(stickiness.clone()),
+                                payload: strategy_variant.payload.clone(),
+                                overrides: None,
```

Comment:
Just checking: are overrides not needed?

Reply:
Yep,  they're specifically excluded in the spec. unleash-types specifies a special strategy variant which holds only the things that are required. We're mapping this to a base variant here so we can use the existing variant flow and intentionally ignoring the override here